src/settingsholder.cpp

```diff
@@ -144,8 +167,10 @@ QString SettingsHolder::getReport() const {
   }                                                                     \
   void SettingsHolder::setter(const type& value) {                      \
     if (!has() || getter() != value) {                                  \
+      maybeSaveInTransaction(key, getter(), value, #getter "Changed",   \
+                             userSettings);                             \
       m_settings.setValue(key, value);                                  \
-      emit getter##Changed(value);                                      \
+      emit getter##Changed();                                           \
```

Comment:
Why remove `value` from the signal?

Reply:
because:
1. we don't use it too much
2. it's complex to "cast" the value when we rollback values